src/org/traccar/events/OverspeedEventHandler.java

```diff
@@ -29,40 +30,95 @@
     public static final String ATTRIBUTE_SPEED_LIMIT = "speedLimit";
 
     private boolean notRepeat;
+    private long minimalDuration;
 
     public OverspeedEventHandler() {
         notRepeat = Context.getConfig().getBoolean("event.overspeed.notRepeat");
+        minimalDuration = Context.getConfig().getLong("event.overspeed.minimalDuration") * 1000;
+    }
+
+    public static Event updateOverspeedState(DeviceState deviceState, double speedLimit,
+            long minimalDuration, boolean notRepeat) {
+        Event result = null;
+        if (deviceState.getOverspeedState() != null && !deviceState.getOverspeedState()
+                && deviceState.getOverspeedPosition() != null && speedLimit != 0) {
+            long currentTime = System.currentTimeMillis();
+            Position overspeedPosition = deviceState.getOverspeedPosition();
+            long overspeedTime = overspeedPosition.getFixTime().getTime();
+            if (overspeedTime + minimalDuration <= currentTime) {
+                result = new Event(Event.TYPE_DEVICE_OVERSPEED, overspeedPosition.getDeviceId(),
+                        overspeedPosition.getId());
+                result.set("speed", overspeedPosition.getSpeed());
+                result.set(ATTRIBUTE_SPEED_LIMIT, speedLimit);
+                deviceState.setOverspeedState(notRepeat);
+                deviceState.setOverspeedPosition(null);
+            }
+        }
+        return result;
+    }
+
+    public static Event updateOverspeedState(DeviceState deviceState, Position position, double speedLimit,
+            long minimalOverspeedDuration, boolean notRepeat) {
+        Event result = null;
+
+        Boolean oldOverspeed = deviceState.getOverspeedState();
+
+        long currentTime = position.getFixTime().getTime();
+        boolean newOverspeed = position.getSpeed() > speedLimit;
+        if (newOverspeed && !oldOverspeed) {
+            if (deviceState.getOverspeedPosition() == null) {
+                deviceState.setOverspeedPosition(position);
+            }
+        } else if (oldOverspeed && !newOverspeed) {
+            deviceState.setOverspeedState(false);
+            deviceState.setOverspeedPosition(null);
+        } else {
+            deviceState.setOverspeedPosition(null);
+        }
+        Position overspeedPosition = deviceState.getOverspeedPosition();
+        if (overspeedPosition != null) {
+            long overspeedTime = overspeedPosition.getFixTime().getTime();
+            if (newOverspeed && overspeedTime + minimalOverspeedDuration <= currentTime) {
```

Comment:
Now it seems like there is some duplication between two methods. Is it possible to share the code? Does it make sense?